spec.html

```diff
@@ -13569,29 +13639,52 @@ <h1>Runtime Semantics: Evaluation</h1>
-          1. Let _propertyKey_ be ? ToPropertyKey(_propertyNameValue_).
           1. If the code matched by this |MemberExpression| is strict mode code, let _strict_ be *true*; else let _strict_ be *false*.
-          1. Return a value of type Reference whose base value component is _bv_, whose referenced name component is _propertyKey_, and whose strict reference flag is _strict_.
+          1. Return ? EvaluateDynamicPropertyAccess(_baseValue_, |Expression|, _strict_).
         </emu-alg>
         <emu-grammar>MemberExpression : MemberExpression `.` IdentifierName</emu-grammar>
         <emu-alg>
           1. Let _baseReference_ be the result of evaluating |MemberExpression|.
           1. Let _baseValue_ be ? GetValue(_baseReference_).
-          1. Let _bv_ be ? RequireObjectCoercible(_baseValue_).
-          1. Let _propertyNameString_ be StringValue of |IdentifierName|.
           1. If the code matched by this |MemberExpression| is strict mode code, let _strict_ be *true*; else let _strict_ be *false*.
-          1. Return a value of type Reference whose base value component is _bv_, whose referenced name component is _propertyNameString_, and whose strict reference flag is _strict_.
+          1. Return ? EvaluateStaticPropertyAccess(_baseValue_, |IdentifierName|, _strict_);
         </emu-alg>
         <emu-grammar>CallExpression : CallExpression `[` Expression `]`</emu-grammar>
-        <p>Is evaluated in exactly the same manner as <emu-grammar>MemberExpression : MemberExpression `[` Expression `]`</emu-grammar> except that the contained |CallExpression| is evaluated in step 1.</p>
+        <emu-alg>
+          1. Let _baseReference_ be the result of evaluating |CallExpression|.
+          1. Let _baseValue_ be ? GetValue(_baseReference_).
+          1. If the code matched by this |CallExpression| is strict mode code, let _strict_ be *true*, else let _strict_ be *false*.
+          1. Return ? EvaluateDynamicPropertyAccess(_baseValue_, |Expression|, _strict_).
+        </emu-alg>
         <emu-grammar>CallExpression : CallExpression `.` IdentifierName</emu-grammar>
-        <p>Is evaluated in exactly the same manner as <emu-grammar>MemberExpression : MemberExpression `.` IdentifierName</emu-grammar> except that the contained |CallExpression| is evaluated in step 1.</p>
+        <emu-alg>
+          1. Let _baseReference_ be the result of evaluating |CallExpression|.
+          1. Let _baseValue_ be ? GetValue(_baseReference_).
+          1. If the code matched by this |CallExpression| is strict mode code, let _strict_ be *true*, else let _strict_ be *false*.
+          1. Return ? EvaluateStaticPropertyAccess(_baseValue_, |IdentifierName|, _strict_);
+        </emu-alg>
       </emu-clause>
     </emu-clause>
 
+    <emu-clause id="sec-evaluate-dynamic-property-access" aoid="EvaluateDynamicPropertyAccess">
+      <h1>Runtime Semantics: EvaluateDynamicPropertyAccess(_baseValue_, _expression_, _strict_ )</h1>
```

Comment:
Hm... I'm not a fan of the Static/Dynamic naming here. Maybe ExpressionPropertyAccess and IdentifierPropertyAccess? @bakkot @ljharb wdyt?

Reply:
"static" and "computed" are the terms I typically see/use

Reply:
I'm fine with that too, though I still prefer the more literal names -- doesn't leave room for misinterpreting what exactly is static or computed.

Reply:
What about `EvaluateExpressionKeyPropertyAccess` and `EvaluateIdentifierKeyPropertyAccess`?

Reply:
I don't like `Identifier` in the name at all. It's an inline string-named property access. And computed property accesses are property accesses where the property name is computed by evaluating a given expression. So `ExpressionPropertyAccess` / `ComputedPropertyAccess` and `StringPropertyAccess` / `StaticPropertyAccess` / `FixedPropertyAccess` are best IMO.

I also don't like `Key` in the name unless it's preceded by `Property`.

Reply:
> Note that, with the advent of private fields and methods, there may be a future `EvaluatePrivatePropertyAccess`.

In which case I think that `Identifier` and `Expression` are probably the best differentiators - because then it can simply be named `EvaluatePrivateNamePropertyAccess` or `EvaluatePrivateNameKeyPropertyAccess`

Reply:
I've gone with the naming I mentioned above, since it feels like a good way to leave the door open to *PrivateName` keys, but I didn't use the `PropertyKey` naming scheme because `PropertyKeyPropertyAccess` gets a bit unwieldy. Long term, if there's an issue, I figure it can always be renamed.